packages/core/src/test/volume-prop.test.ts

```diff
@@ -0,0 +1,45 @@
+import {evaluateVolume} from '../volume-prop';
+import {expectToThrow} from './expect-to-throw';
+
+describe('EvaluateVolume does not throw', () => {
+	test('Volume is undefined', () => {
+		const toEvaluate = {frame: 10, volume: undefined};
+		expect(evaluateVolume(toEvaluate)).toEqual(1);
+	});
+	test('Volume is smaller than one', () => {
+		const smallVolume = 0.5;
+		const toEvaluate = {frame: 10, volume: smallVolume};
+		expect(evaluateVolume(toEvaluate)).toEqual(smallVolume);
+	});
+	test('Volume is bigger than one', () => {
+		const toEvaluate = {frame: 10, volume: 10};
+		expect(evaluateVolume(toEvaluate)).toBe(1);
+	});
+	test('evaluated volume from frame that is smaller than one', () => {
+		const smallFrame = 0.5;
+		const toEvaluate = {frame: smallFrame, volume: (frame: number) => frame};
+		expect(evaluateVolume(toEvaluate)).toBe(smallFrame);
+	});
+	test('evaluated volume from frame that is bigger than one', () => {
+		const toEvaluate = {frame: 10, volume: (frame: number) => frame};
+		expect(evaluateVolume(toEvaluate)).toBe(1);
+	});
+});
+
+describe('evaluateVolume throws exception', () => {
+	test('It should throw if volume prop is neither number nor undefined', () => {
+		const toEvaluate = {frame: 10, volume: 'NaN'};
+		expectToThrow(() => {
+			// @ts-expect-error
+			evaluateVolume(toEvaluate);
+		}, /volume is not a function/);
+	});
+	test('It should throw if frame is invalid type', () => {
+		const invalidFrame = 'NaN';
+		const toEvaluate = {frame: invalidFrame, volume: (frame: any) => frame};
```

Comment:
I refactored this portion a bit.
Two possible cases are imagineable, NaN and string, so I split it up.

Frame will never be 'NaN', but what can be tested is that the user returns NaN even though the frame is a number. I'm changes the test and removed the `any` also